test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -184,6 +262,53 @@ public void windowClosing(WindowEvent e) {
         windowList.add(frame);
     }
 
+    private static void captureScreen(Rectangle bounds) {
+        if (robot == null) {
+            try {
+                robot = new Robot();
+            } catch (AWTException e) {
+                String errorMsg = "Failed to create the " +
+                        "instance of Robot.";
+                JOptionPane.showMessageDialog(frame, errorMsg, "Failed",
+                        JOptionPane.ERROR_MESSAGE);
+                forceFail(errorMsg + e.getMessage());
+            }
+        }
+        MultiResolutionImage multiResolutionImage = robot
+                .createMultiResolutionScreenCapture(bounds);
+        List<Image> imageList =
+                multiResolutionImage.getResolutionVariants();
+        Image image = imageList.get(imageList.size() - 1);
+        File file = new java.io.File("CaptureScreen_" +
+                imgCounter.incrementAndGet() + ".png");
+        try {
+            ImageIO.write((RenderedImage) image, "png", file);
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
+        }
+    }
+
+    private static void captureScreen(int selectedIndex) {
+        if (selectedIndex == 0) {
+            Arrays.stream(GraphicsEnvironment.getLocalGraphicsEnvironment()
+                    .getScreenDevices())
+                    .map(GraphicsDevice::getDefaultConfiguration)
+                    .map(GraphicsConfiguration::getBounds)
+                    .forEach(PassFailJFrame::captureScreen);
+        }
+
+        if (selectedIndex == 1) {
+            windowList.stream()
+                    .filter(Window::isShowing)
+                    .map(Window::getBounds)
+                    .forEach(PassFailJFrame::captureScreen);
+        }
+
+        JOptionPane.showMessageDialog(frame, "Screen Captured " +
+                        "Successfully", "Screen Capture",
+                JOptionPane.INFORMATION_MESSAGE);
```

Comment:
```suggestion
        JOptionPane.showMessageDialog(frame,
                "Screen captured successfully",
                "Screen Capture",
                JOptionPane.INFORMATION_MESSAGE);
```
It's better not to split strings unnecessarily. The messages usually use regular capitalisation as opposed to title capitalisation.

Reply:
@aivanov-jdk, @lawrence-andrew 

> If the displayed message isn't modal, like a tooltip or a notification which disappears automatically after a certain time, it could be better appreciated. But it's harder to implement.

We could use Timer for disappearing messages, but might be a overkill here.

```
  JOptionPane pane = new JOptionPane("Screen capture successful", JOptionPane.INFORMATION_MESSAGE);
  final JDialog dialog = pane.createDialog("Screen Capture");
  Timer timer = new Timer(2000, e -> dialog.dispose());
  timer.setRepeats(false);
  timer.start();
  dialog.setVisible(true);
```

Reply:
> > If the displayed message isn't modal, like a tooltip or a notification which disappears automatically after a certain time, it could be better appreciated. But it's harder to implement.
> 
> We could use Timer for disappearing messages, but might be an overkill here.

How else could one implement a disappearing message?

This approach still shows a *modal* dialog box. While it's shown the user can't do anything but dispose the dialog.

In my opinion, if we show the dialog, let the user to dismiss it.

The message confirming screenshots are successfully taken should be non-intrusive, it may be displayed in a small (translucent) window close to the button which will automatically disappear after a specified timeout. An option to dismiss it before the timeout is even better. Yet it shouldn't prevent access to other UI elements so that the tester could click Fail right away.

However, I may be overthinking it… Implementing such a UI is likely an overkill.

Yet I'm still not convinced a modal dialog box is needed. Consider seeing a modal dialog box that says "File has been saved successfully" each time you select Save command in a text editor or any other editor. You expect it to succeed, thus notification is need only if an error occurred. As the result of the error message, you'll try saving the file again.